### ISSUE TITLE
Locking phone while moving around map causes crash on iOS

### ISSUE THREAD

**Issue by range-of-motion** [react-native-mapbox-gl/maps#1350]
**Describe the bug**  
Locking phone while moving around map (before animation has finished) causes crash on iOS. I think it's fairly self-explanatory, but I'm happy to provide additional information if need be.

**Versions**  
 - Platform: iOS
 - Platform OS: `14.x`
 - Device: iPhone (any)
 - Emulator/ Simulator: no
 - Dev OS: macOS (any)
 - react-native-mapbox-gl version: `8.2.0-beta1`
 - Mapbox GL version: `5.9.0`
 - React Native version: `0.62.3`

**Additional information**
Here's a screenshot of one of the crashes as provided by App Store Connect.

![image](https://user-images.githubusercontent.com/9268822/118401438-02832280-b666-11eb-9912-8a7afbe4eed3.png)

**Comment by range-of-motion** [comment]
I think [this is related](https://github.com/mapbox/mapbox-gl-native-ios/issues/553).

**Comment by range-of-motion** [comment]
Any word about this, or the upgrade to Mapbox's SDK for iOS v6.3?

**Comment by range-of-motion** [comment]
No worries, I know how it is. I appreciate the work you put into this.

I could try MapLibre, but I have to admit I'm unfamiliar with it. Do you mean I should just open up a native iOS project in Xcode, display a map using MapLibre's SDK and see if it suffers from the same bug?

**Comment by ferdicus** [comment]
MapLibre is just a fork of Mapbox before they changed their licensing.  
It is a drop in replacement for the SDKs that we're currently using in `8.2.0-x`. 

Steps on how to use it are explained within our installation guide.

**Comment by range-of-motion** [comment]
Couldn't figure out how to use MapLibre, but that's probably my stupidity. 

I *did* spend the night trying to figure out which iOS SDK versions have this bug, and which do not. 

| Mapbox iOS SDK version | Has this bug? |
| - | - |
| 5.9.x | Yes |
| 6.0.x | Yes |
| 6.1.x | Yes |
| 6.2.x | No |
| 6.3.x | No |

TLDR–since 6.2 the bug has disappeared.

**Comment by range-of-motion** [comment]
Say you agree with me (and want to upgrade to 6.2), can you shed some light on the ETA for such an upgrade?

*And is there any other way to contact you besides these issues? I highly depend on this.. 🙄*

**Comment by ferdicus** [comment]
> Say you agree with me (and want to upgrade to 6.2), can you shed some light on the ETA for such an upgrade?
> 


You can check the [iOS installation guide](https://github.com/react-native-mapbox-gl/maps/blob/master/ios/install.md#mapbox-maps-sdk) - it should be descriptive enough to get you going.

> _And is there any other way to contact you besides these issues? I highly depend on this.. 🙄_

I'd prefer it to be in the issues, as other people may have the same questions - so it's documented.

**Comment by range-of-motion** [comment]
> You can check the iOS installation guide - it should be descriptive enough to get you going.

Thanks, didn't know it was that easy. I successfully used 6.2 and was able to get rid of the crash upon locking or entering the background. 

I'm a bit skeptical about upgrading like this however, are there any unexpected side-effects or bugs that could appear because I'm a major version up from the library's current iOS SDK version?

**Comment by ferdicus** [comment]
> I'm a bit skeptical about upgrading like this however, are there any unexpected side-effects or bugs that could appear because I'm a major version up from the library's current iOS SDK version?

I mean, there can always be bugs, right?  
Mapbox was confident enough to release `6.2` as production ready, so there's that.

**Comment by range-of-motion** [comment]
Right, but I'm guessing there is somewhat of a bridge between this package and the iOS SDK? I was wondering if there would be any differences in that "bridge" between 5.x.x and 6.x.x?

**Comment by ferdicus** [comment]
> Right, but I'm guessing there is somewhat of a bridge between this package and the iOS SDK? I was wondering if there would be any differences in that "bridge" between 5.x.x and 6.x.x?

it's a bit complicated - we are not developing in lockstep with the native SDKs.   
For example `8.2.0-beta2` can be used with the default packed in iOS SDK `5.9.0` or with the last supported `6.3.0`.
The apis exposed by `5.9.0` shouldn't have changed towards `6.3.0`, however there might have been added some new features/ some bugs might've been fixed. 

If it makes you feel better, we've been using `8.2.0-beta2` with `6.3.0` in production on iOS and have so far a positive experience. I think it depends on your usage/ usecase - your mileage might vary.

**Comment by range-of-motion** [comment]
Thanks, that's reassuring. 

I took `8.2.0-beta2` and `6.3.0` in production since yesterday as well, and so far it seems positive. That'll be it for now, I guess. Thanks for everything! 🙂

**Comment by kbaxx** [comment]
Experiencing the same issue here on our production mobile apps having a major impact on our business. 

@ferdicus Is there an ETA on getting the fix also into this package or does this solely depend on the connected mapbox ios sdk issue? Thanks in advance.

@range-of-motion How is the use of a different source resolving the issue on this package though? Gives a bit of a wrong impression of this issue imho.

**Comment by range-of-motion** [comment]
There is no bug in this package. The bug is in Mapbox's iOS SDK. 

As you can see in the table I posted a few comments above, it's fixed since 6.2. However this package still uses 5.9. Hence why you're experiencing this.

**Comment by kbaxx** [comment]
@ferdicus The residing open question is then: Are you planning to switch to 6.2 soon for this package as well?

**Comment by ferdicus** [comment]
@kbaxx, please read the whole thread carefully and then see if that answers your question.  
I've already taken the time to explain how one can replace the native sdks.

**Comment by mvanroon** [comment]
Looks like this was fixed in [Maplibre 5.12.0](https://github.com/maplibre/maplibre-gl-native/releases/tag/ios-v5.12.0) and [Mapbox v6.2.0](https://github.com/mapbox/mapbox-gl-native-ios/releases/tag/ios-v6.2.0).

Maplibre issue: https://github.com/maplibre/maplibre-gl-native/issues/70
Mapbox issue: https://github.com/mapbox/mapbox-gl-native-ios/issues/440
